conans/test/integration/metadata/test_metadata_collect.py

```diff
@@ -0,0 +1,201 @@
+    print_graph_packages(deps_graph)
+
+    # Installation of binaries and consumer generators
+    conan_api.install.install_binaries(deps_graph=deps_graph, remotes=remotes)
+
+    # Downloading metadata
+    metadata = args.metadata if args.metadata else ["*"]
+    # TODO: This InstallGraph API is EXPERIMENTAL and will change in future versions
+    install_graph = InstallGraph(deps_graph)
+    install_order = install_graph.install_order(flat=True)
+
+    if args.metadata_remote:
+        out = ConanOutput()
+        out.title(f"Downloading metadata from {args.metadata_remote}")
+        remote = conan_api.remotes.get(args.metadata_remote)
+
+        for install_reference in install_order:
+            if install_reference.ref.revision is None:  # Is an editable/system, do not download
+                continue
+            try:
+                conan_api.download.recipe(install_reference.ref, remote, metadata)
+            except ConanException as e:
+                out.warning(f"Recipe {install_reference.ref} not found in remote: {e}")
+                continue
+            for package in install_reference.packages.values():
+                try:
+                    conan_api.download.package(package.pref, remote, metadata)
+                except ConanException as e:
+                    out.warning(f"Package {package.pref} not found in remote: {e}")
+
+    # Copying and collecting metadata from all packages into local copy
+    def _copy_metadata(src, dst):
+        for root, subfolders, files in os.walk(src):
+            relative_path = os.path.relpath(root, src)
+            for f in files:
+                relative_name = os.path.normpath(os.path.join(relative_path, f))
+                if any(fnmatch.fnmatch(relative_name, m) for m in metadata):
+                    os.makedirs(os.path.dirname(os.path.join(dst, relative_name)), exist_ok=True)
+                    shutil.copy2(os.path.join(src, relative_name), os.path.join(dst, relative_name))
+
+    for install_reference in install_order:
+        conanfile = install_reference.node.conanfile
+        if conanfile.recipe_metadata_folder is None:
+            continue
+        folder = os.path.join(output_folder, conanfile.ref.name, str(conanfile.ref.version))
+        if os.path.exists(folder):
+            conanfile.output.warning(f"Folder for {conanfile} already exist, removing it: {folder}")
+            shutil.rmtree(folder)
+        conanfile.output.info(f"Copying recipe metadata from {conanfile.recipe_metadata_folder}")
+        _copy_metadata(conanfile.recipe_metadata_folder, os.path.join(folder, "recipe"))
+        for package in install_reference.packages.values():
+            pkg_metadata_folder = package.conanfile.package_metadata_folder
+            conanfile.output.info(f"Copying package metadata from {pkg_metadata_folder}")
+            _copy_metadata(pkg_metadata_folder, os.path.join(folder, "package"))
+'''
```

Comment:
Perhaps it's unnecessary now, but it could be great to save this piece of code as a Python module in `assets` and copy and paste that file when the tests start. It'd make our lives easier when reading the diff.

Reply:
This is very specific for this test, not like a general asset, isn't it?